testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -162,6 +162,28 @@ def setUp(self):
         self.values['acceptor_resnm'] = np.array([], dtype="<U3")
 
 
+def build_parameters():
```

Comment:
Class attrs don't need to be accessed via `self ` when in the class body. They're already in the namespace. Anyway, I think the solution of having the iterator call within the decorator is the best, as it clearer what's happening and allows customization in place.